atoma-state/src/state_manager.rs

```diff
@@ -1509,23 +1522,32 @@ impl AtomaState {
         stack_small_id: i64,
         estimated_total_compute_units: i64,
         total_compute_units: i64,
-    ) -> Result<()> {
+    ) -> Result<UpdateStackNumComputeUnitsAndClaimFunds> {
         let result = sqlx::query(
-            "UPDATE stacks
+            "UPDATE stacks s
             SET already_computed_units = already_computed_units - ($1 - $2)
-            WHERE stack_small_id = $3",
+            WHERE s.stack_small_id = $3
+            RETURNING 
+                CAST(s.already_computed_units AS FLOAT) / CAST(s.num_compute_units AS FLOAT),
+                s.already_computed_units,
+                (s.is_confidential = true) as is_confidential",
```

Comment:
```suggestion
                CAST(s.already_computed_units AS FLOAT) / CAST(s.num_compute_units AS FLOAT) as ratio,
                s.already_computed_units as stack_computed_units,
                (s.is_confidential = true) as is_confidential",
```
and then deriving the `FromRow` (other comment)